src/xrpld/ledger/detail/View.cpp

```diff
@@ -424,20 +486,38 @@ accountHolds(
         // operation
         if (zeroIfUnauthorized == ahZERO_IF_UNAUTHORIZED)
         {
-            auto const sleIssuance =
-                view.read(keylet::mptIssuance(mptIssue.getMptID()));
-
-            // if auth is enabled on the issuance and mpt is not authorized,
-            // clear amount
-            if (sleIssuance && sleIssuance->isFlag(lsfMPTRequireAuth) &&
-                !sleMpt->isFlag(lsfMPTAuthorized))
+            if (auto const err = requireAuth(view, mptIssue, account);
```

Comment:
This has to be amendment gated. There is a difference between pre/post SAV. For instance, in pre-SAV `amount` is cleared if `sleIssuance` is not null and MPToken is not authorized. Post-SAV clears `amount` is `sleIssuance` doesn't exist.

Reply:
Thank youm, done.